docs/src/compatibility.md

```diff
@@ -104,3 +104,20 @@ PkgB = "≥ 1.2.3"  # [1.2.3,  ∞)
 PkgC = "= 1.2.3"  # [1.2.3, 1.2.3]
 PkgD = "< 1.2.3"  # [0.0.0, 1.2.2]
 ```
+
+### Hyphen specifiers
+
+Hyphen syntax can also be used to specify version ranges:
+
+```toml
+[compat]
+PkgA = "0.1 - 0.2.3"   # [0.1, 0.2.3]
```

Comment:
Given https://discourse.julialang.org/t/how-can-we-encourage-julia-package-developers-to-release-version-1-0-0/29124, maybe one way is to use post 1.0 numbers in examples like this? :smile: At least in addition, to the current example.